File: src/types/transaction.ts
import { Category } from './category';

export interface Transaction {
  id: string; // unique id
  entered: string; // ISO date string for when the transaction was entered
  transactionDate: string; // ISO date string for when the transaction occurred
  paidBy?: string; // Who paid for this expense (defaults to enteredBy)
  owedBy?: string; // Who owes money for this expense
  payTo?: string; // Deprecated - kept for backward compatibility
  amount: number;
  note: string;
  category: string;
  paid: boolean;
  paidDate?: string; // ISO date string, optional
  enteredBy: string; // who entered the transaction
  optional: boolean; // indicates if transaction is optional
  inSettlement?: boolean; // whether this transaction is part of a settlement
  categories?: Category[]; // array of categories associated with this transaction
}

File: backend/models/transaction.go
package models

import "time"

type Transaction struct {
	ID              string     `json:"id"`
	Amount          float64    `json:"amount"`
	Description     string     `json:"description"`
	Date            time.Time  `json:"date"`
	TransactionDate time.Time  `json:"transactionDate"`
	Type            string     `json:"type"`
	PaidBy          string     `json:"paidBy,omitempty"` // User ID of who paid for this expense
	OwedBy          string     `json:"owedBy,omitempty"` // User ID of who owes money for this expense
	PayTo           string     `json:"payTo,omitempty"`  // Deprecated - kept for backward compatibility
	Paid            bool       `json:"paid"`
	PaidDate        string     `json:"paidDate,omitempty"`
	EnteredBy       string     `json:"enteredBy"`
	Optional        bool       `json:"optional"`
	UserID          string     `json:"userId,omitempty"`
	Note            string     `json:"note,omitempty"`
	InSettlement    bool       `json:"inSettlement,omitempty"` // Whether this transaction is part of a settlement
	Categories      []Category `json:"categories,omitempty"`

	// Display names - populated by API for frontend display
	PaidByName    string `json:"paidByName,omitempty"`
	OwedByName    string `json:"owedByName,omitempty"`
	EnteredByName string `json:"enteredByName,omitempty"`
}

// TransactionCategory represents a join between a transaction and a category
type TransactionCategory struct {
	ID            int       `json:"id"`
	TransactionID string    `json:"transactionId"`
	CategoryID    int       `json:"categoryId"`
	Amount        float64   `json:"amount"`
	CreatedAt     time.Time `json:"createdAt"`
}
